### ISSUE TITLE
block_collapse sometimes fails with block matrices with ZeroMatrix entries (v. 1.4)

### ISSUE THREAD

**Issue by syanga** [sympy/sympy#18002]
For example, if I have:

`n = symbols('n')`
`A = MatrixSymbol('A', n, n)`
`B = MatrixSymbol('B', n, n)`
`C = BlockMatrix([[ZeroMatrix(n,n), ZeroMatrix(n,n)], [ZeroMatrix(n,n), A]])`
`D = BlockMatrix([[ZeroMatrix(n,n), ZeroMatrix(n,n)], [ZeroMatrix(n,n), B]])`

The expression `C*D` does not throw an error, but block_collapse(C*D) fails with:
`AttributeError: 'Zero' object has no attribute 'is_Identity'`

Is this intended?

**Comment by oscarbenjamin** [comment]
This is a bug in SymPy 1.4 that is fixed in master and (not quite released) 1.5.

You can either use master or install the prerelease of 1.5 with `pip install --pre sympy`.

```julia
In [1]: n = symbols('n') 
   ...: A = MatrixSymbol('A', n, n) 
   ...: B = MatrixSymbol('B', n, n) 
   ...: C = BlockMatrix([[ZeroMatrix(n,n), ZeroMatrix(n,n)], [ZeroMatrix(n,n), A]]) 
   ...: D = BlockMatrix([[ZeroMatrix(n,n), ZeroMatrix(n,n)], [ZeroMatrix(n,n), B]])                                               

In [2]: block_collapse(C*D)                                                                                                       
Out[2]: 
⎡𝟘   𝟘 ⎤
⎢      ⎥
⎣𝟘  A⋅B⎦
```

**Comment by syanga** [comment]
Got it, thanks!
